File: samples/android/java/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import smjni.jnigen.ExposeToNative;

@ExposeToNative
public class MainActivity extends AppCompatActivity {

    static {
       System.loadLibrary("myapplication");
    }

    private native int nativeFunction(byte b);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        nativeFunction((byte)0);
    }
}

File: samples/android/kotlin-ksp-kts/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import smjni.jnigen.ExposeToNative

@ExposeToNative
class MainActivity : AppCompatActivity() {

    companion object {
        init {
            System.loadLibrary("myapplication")
        }
    }

    private external fun nativeFunction(b: Byte): Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        nativeFunction(0)
    }
}